src/components/component-playground.jsx

```diff
@@ -525,5 +548,40 @@ module.exports = React.createClass({
       orientation: contentNode.offsetHeight > contentNode.offsetWidth ?
                    'portrait' : 'landscape'
     });
+  },
+
+  _getFilteredComponents() {
+    return _.mapValues(this.props.components, function(fixtures) {
+      return _.assign({}, fixtures, {
+      // Only select fixtures matching search text and rebuild the object.
+        fixtures: _.pick(fixtures.fixtures,
+            function(fixtureProps, fixtureName) {
+              return fixtureName.indexOf(this.state.searchText) !== -1;
+            }.bind(this)
+        )
+      });
+    }.bind(this));
+  },
+
+  _getFilteredFixtures() {
+    // Get components which match search text by one or more fixture names.
+    var matchedByFixtureName = _.pick(this._getFilteredComponents(),
+        // Ignore the component if no fixture matched.
+        function(fixtures) {
+          return !_.isEmpty(fixtures.fixtures);
+        }
+    );
+
+    // Get components which match search text by component name and the
+    // currently selected one.
+    var matchedByComponentName = _.pick(this.props.components,
+        function(fixtureProps, componentName) {
+          return componentName.indexOf(this.state.searchText) !== -1 ||
+                 componentName === this.props.component;
```

Comment:
I'm assuming that's because the sidebar iterates through the keys in the `components` object assuming that it will get them in an alphabetical order. Object property iteration order is not guaranteed, so we should change the logic to first sort the keys, and then iterate them.